libs/ledger-live-common/src/families/ethereum/__tests__/signOperation.test.ts

```diff
@@ -0,0 +1,141 @@
+import { ethers } from "ethers";
+import { BigNumber } from "bignumber.js";
+import {
+  cryptocurrenciesById,
+  getCryptoCurrencyById,
+} from "@ledgerhq/cryptoassets/currencies";
+
+import createTransaction from "../createTransaction";
+import { fromAccountRaw } from "../../../account";
+import { ethereum1 } from "../datasets/ethereum1";
+import { signOperation } from "../signOperation";
+import { setEnv } from "../../../env";
+import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
+
+const signTransaction = jest.fn(() => {
+  return Promise.resolve({
+    r: "r",
+    s: "s",
+    v: "v",
+  });
+});
+
+jest.mock("@ledgerhq/hw-app-eth", () => {
+  return {
+    ...jest.requireActual("@ledgerhq/hw-app-eth"),
+    default: class {
+      signTransaction = signTransaction;
+      setLoadConfig = () => {};
+    },
+  };
+});
+
+jest.mock("../../../hw/deviceAccess", () => ({
+  withDevice: () => (job) => job({ decorateAppAPIMethods: () => {} }),
+}));
+
+const dummyAccount = fromAccountRaw(ethereum1);
+
+const currencies = Object.values(cryptocurrenciesById).filter((currency) => {
+  return currency.ethereumLikeInfo;
+});
+
+describe("signOperation", () => {
+  describe("chainId encoding (EIP155)", () => {
+    describe("Transaction type 2", () => {
+      const type2Currencies: CryptoCurrency[] = [
+        "ethereum",
+        "ethereum_goerli",
+        "polygon",
+      ].map(getCryptoCurrencyById);
+
+      type2Currencies.forEach((currency) => {
+        beforeEach(() => {
+          signTransaction.mockImplementation(() => {
```

Comment:
I'd rather use `mockImplementationOnce` here, so we don't have false positive from a previous loop, just in case

Reply:
the test is failing with the `mockImplementationOnce`, I'll check it out